File: app/src/main/java/com/stone/stonemusic/net/ResponseHandler.java
package com.stone.stonemusic.net;

/**
 * @Author: stoneWang
 * @CreateDate: 2019/8/10 22:26
 * @Description: 请求的回调
 * 声明泛型RESPONSE
 */
public interface ResponseHandler<RESPONSE> {

    /**
     * 失败的回调
     */
    void onError(int type, String message);

    /**
     * 成功的回调
     * 泛型：RESPONSE
     */
    void onSuccess(int type, RESPONSE result);

}

File: app/src/main/java/com/stone/stonemusic/net/ResponseHandler.kt
package com.stone.stonemusic.net

/**
 * @Author: stoneWang
 * @CreateDate: 2019/8/5 10:10
 * @Description: 请求的回调
 */
interface ResponseHandler<RESPONSE> {
    /**
     * 成功的回调
     */
    fun onError(type:Int, msg: String?)

    /**
     * 失败的回调
     * 泛型：RESPONSE
     */
    fun onSuccess(type: Int, result: RESPONSE)
}